packages/components/src/custom-select-control/test/index.js

```diff
@@ -87,4 +88,127 @@ describe( 'CustomSelectControl', () => {
 			screen.getByRole( 'button', { name: 'Custom select' } )
 		).toHaveTextContent( 'Hint' );
 	} );
+
+	it( 'Initial value should be replaced when a new item is selected', async () => {
+		const user = userEvent.setup();
+		const mock = jest.fn();
+
+		render(
+			<CustomSelectControl
+				onChange={ ( { selectedItem } ) => mock( selectedItem ) }
+				options={ options }
+				__nextUnconstrainedWidth
+			/>
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+
+		expect( currentValue ).toHaveTextContent( /violets/i );
+
+		await user.click( currentValue );
+
+		const newValue = screen.getByRole( 'option', {
+			name: /poppy/i,
+		} );
+
+		await user.click( newValue );
+
+		expect( currentValue ).toHaveTextContent( /poppy/i );
+		expect( mock ).toHaveBeenCalled();
```

Comment:
Isn't this testing an implementation detail? Isn't it sufficient to check the text content, and what I'd add, the `value` of the selected item?

Reply:
Which leads me to a suggestion — we should be probably checking for `onChange` to have been called (or not) after every interaction. `toHaveBeenCalledTimes( n )` and `toHaveLastBeenCalledWith( x )` can be helpful in this case.

Reply:
I'm not sure how we can test for the `onChange` mock now that we have controlled and uncontrolled tests. So it's been removed for now. If there is a way and it's beneficial still, I might need a bit of help there.